edb/common/ast/base.py

```diff
@@ -192,30 +192,9 @@ def __init__(self, **kwargs):
                 f'cannot instantiate abstract AST node '
                 f'{self.__class__.__name__!r}')
 
-        object.__setattr__(self, 'parent', None)
+        object.__setattr__(self, 'parent', kwargs.get('parent', None))
         self._init_fields(kwargs)
```

Comment:
Can you inline this function? It's not used anywhere else but `AST.__init__` is called a lot. Inlining will shave the calls number a bit and is completely safe in this case (even though this is a micro opt).

Reply:
I did but this is a miniscule difference since there is just one call per `__init__` call. It's less readable for dubious benefit.